R/create_default_parameters.R

```diff
@@ -602,6 +602,57 @@ create_default_recruitment <- function(
 #' @seealso
 #' * [create_default_parameters()]
 #' @export
+#' @examples
+#' \dontrun{
+#' # Prepare the package data for being used in a FIMS model
+#' data("data1")
+#' fims_frame <- FIMSFrame(data1)
+#'
+#' # Define the fleet specifications 
+#' fleet1 <- survey1 <- list(
+#'   selectivity = list(form = "LogisticSelectivity"),
+#'   data_distribution = c(
+#'     Index = "DlnormDistribution",
+#'     AgeComp = "DmultinomDistribution"
+#'   )
+#' )
+#' fleet2 <- list(
+#'   selectivity = list(form = "DoubleLogisticSelectivity"),
+#'   data_distribution = c(
+#'     Index = "DlnormDistribution",
+#'     AgeComp = "DmultinomDistribution",
+#'     LengthComp = "DmultinomDistribution"
+#'   )
+#' )
+#'
+#' # Create defaults and adjust the fishing mortality from its default values
+#' default_parameters <- fims_frame |>
+#'   create_default_parameters(
+#'     fleets = list(fleet1 = fleet1, fleet2 = fleet2, survey1 = survey1)
+#'   )
+#' updated_parameters <- default_parameters |>
+#'   update_parameters(
+#'      modified_parameters = list(
+#'        fleet1 = list(
+#'          Fleet.log_Fmort.value = log(c(
+#'            0.009459165, 0.027288858, 0.045063639,
+#'            0.061017825, 0.048600752, 0.087420554,
+#'            0.088447204, 0.186607929, 0.109008958,
+#'            0.132704335, 0.150615473, 0.161242955,
+#'            0.116640187, 0.169346119, 0.180191913,
+#'            0.161240483, 0.314573212, 0.257247574,
+#'            0.254887252, 0.251462108, 0.349101406,
+#'            0.254107720, 0.418478117, 0.345721184,
+#'            0.343685540, 0.314171227, 0.308026829,
+#'            0.431745298, 0.328030899, 0.499675368
+#'          ))
+#'        )
+#'      )
+#'   )
+#' # purrr::map_vec() can be used to compare the length of adjusted parameter vectors with defaults
+#' default_fleet1 <- purrr::map_vec(default_parameters[["parameters"]][["fleet1"]], \(x) length(x))
+#' updated_fleet1 <- purrr::map_vec(updated_parameters[["parameters"]][["fleet1"]], \(x) length(x))
+#' }
```

Comment:
What do you think about also adding the following code:
```
purrr::map_df(
  updated_parameters[["parameters"]], \(x) purrr::map_vec(x, length),
  .id = "module"
) |>
  tibble::column_to_rownames(var = "module") |>
  t()
```
which shows something like this
![image](https://github.com/user-attachments/assets/8d7f018e-eb22-4952-8633-a2ba6c156fac)

Reply:
It looks nice! I will add it to both the example and the vignette.

Reply:
Done!